File: src/test/java/com/mparticle/kits/mocks/MockContextApplication.java
package com.mparticle.kits.mocks;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

public class MockContextApplication extends Application {
    @Override
    public Context getApplicationContext() {
        return this;
    }

    @Override
    public SharedPreferences getSharedPreferences(String name, int mode) {
        return new MockSharedPreferences();
    }

    @Override
    public void registerActivityLifecycleCallbacks(ActivityLifecycleCallbacks callback) {
        //do nothing
    }
}

File: src/test/java/com/mparticle/kits/mocks/MockContextApplication.kt
package com.mparticle.kits.mocks

import android.app.Application
import android.content.SharedPreferences
import com.mparticle.kits.mocks.MockSharedPreferences
import android.app.Application.ActivityLifecycleCallbacks
import android.content.Context

class MockContextApplication : Application() {
    override fun getApplicationContext(): Context {
        return this
    }

    override fun getSharedPreferences(name: String, mode: Int): SharedPreferences {
        return MockSharedPreferences()
    }

    override fun registerActivityLifecycleCallbacks(callback: ActivityLifecycleCallbacks) {
        //do nothing
    }
}
